test/parallel/test-http-keepalive-override.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const http = require('http');
+
+let keepSocketAliveCalled = 0;
+let reuseSocketCalled = 0;
+
+const server = http.createServer((req, res) => {
+  res.end('ok');
+}).listen(0, common.mustCall(() => {
+  const agent = http.Agent({
+    keepAlive: true,
+    maxSockets: 5,
+    maxFreeSockets: 2
+  });
+
+  const keepSocketAlive = agent.keepSocketAlive;
+  const reuseSocket = agent.reuseSocket;
+
+  let expectedSocket;
+
+  agent.keepSocketAlive = (socket) => {
```

Comment:
Wrap this in `common.mustCall(...., 3)` and remove the two unnecessary assertions (one in here and at the end)?